### PR TITLE
Add Option to control the injection of provider-id to metadata

### LINKED ISSUE #345: BUG: provider-id should be opt-in
With #292 a breaking change was introduced, which forces `provider-id: proxmox://{{ .InstanceID }}` through cloud-init: https://github.com/ionos-cloud/cluster-api-provider-proxmox/blob/51a9d0a454937e3a12f74cbdf287afe10756e1b7/pkg/cloudinit/metadata.go#L23C1-L23C12

This will break any setup, that requires a `providerID` in a different format than set by CAPMOX, and it can't be altered by a third party CCM/automation either, since `Node.spec.providerID` is an immutable field:
```
$ k patch node dev-node-f56ef5 -p '{"spec":{"providerID":"foo"}}'
The Node "dev-node-f56ef5" is invalid: spec.providerID: Forbidden: node updates may not change providerID except from "" to valid
```

Slightly related to #342.

### PR DESCRIPTION
**Pull request by mcbenjemaa**
This PR:

- Adds an Option to control whether to inject the `providerID` to metadata or not.

fix #345

### CODE REVIEW

File: api/v1alpha1/proxmoxmachine_types.go [hunk 1]
```diff
@@ -106,7 +106,13 @@ type ProxmoxMachineSpec struct {
 	// +kubebuilder:validation:XValidation:rule="self.end >= self.start",message="end should be greater than or equal to start"
 	VMIDRange *VMIDRange `json:"vmIDRange,omitempty"`
 
+	// Checks defines possibibles checks to skip.
```
Comment (wikkyk) [inline api/v1alpha1/proxmoxmachine_types.go:109]: typo

Comment (mcbenjemaa) [reply api/v1alpha1/proxmoxmachine_types.go:109]: Suggested by Copilot, you can submit an issue 😄

File: api/v1alpha1/proxmoxmachine_types.go [hunk 2]
```diff
@@ -497,6 +503,13 @@ type VMIDRange struct {
 	End int64 `json:"end"`
 }
 
+// MetadataSettings defines the metadata settings for the machine.
+type MetadataSettings struct {
+	// ProviderIDInjection enables the injection of the providerID into the cloudinit metadata.
+	// +optional
+	ProviderIDInjection bool `json:"providerIDInjection,omitempty"`
```
[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply api/v1alpha1/proxmoxmachine_types.go:510]: You think so?

[the remark this answers is not in the document]
Comment (mcbenjemaa) [reply api/v1alpha1/proxmoxmachine_types.go:510]: let's make the users to add this providerID so they know what their are doing.

[the remark this answers is not in the document]
Comment (wikkyk) [reply api/v1alpha1/proxmoxmachine_types.go:511]: I don't really have an opinion, I was just wondering which option would be better. Perhaps this is something that doesn't need to be enabled by default.

**Review by wikkyk — APPROVED** [review]
I'll let the missing tests slide as this is a bit of an emergency fix.

File: api/v1alpha1/proxmoxmachine_types.go (hunk 1 above)
Comment (wikkyk) [reply api/v1alpha1/proxmoxmachine_types.go]: you need to regen.

**Comment by mcbenjemaa** [comment]
Closes https://github.com/ionos-cloud/cluster-api-provider-proxmox/issues/342

**Review by wikkyk — APPROVED** [review]

**Pull request merged**